Add columns regarding Value Ref in expconf

mrosanes opened the pull request:
Add two new columns in expconf widget:
- Ref Enabled:
Channel value referencing active or not

- Ref Pattern:
Channel value referencing pattern

Commit:
Add columns regarding Value Ref in expconf

Add two new columns in expconf widget:
- Ref Enabled:
Channel value referencing active or not

- Ref Pattern:
Channel value referencing pattern

reszelaz reviewed (changes requested):
expconf problems:
* data fields are not populated with the values comming from the configuration
* Ref Enabled editor is string and not bool (True/False)
* when leaving edition mode of Ref Enabled or Ref Pattern with empty data field we get the following error:
```
Traceback (most recent call last):
  File "/home/zreszela/workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/measurementgroup.py", line 602, in setData
    item.setData(index, qvalue)
  File "/home/zreszela/workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/measurementgroup.py", line 361, in setData
    raise NotImplementedError('Unknown role')
NotImplementedError: Unknown role
```

reszelaz commented:
How to modifye configuration without expconf:
```python 
import json
conf = json.loads(mntgrp03_sep2.configuration)
conf["controllers"]["tango://pc255.cells.es:10000/controller/dummytwodcontroller/twodctrl03]["channels"]["tango://pc255.cells.es:10000/expchan/twodctrl03/1"]["value_ref_pattern"] = "h5file:///tmp/test_mg_try.h5::dataset_check_me"
mntgrp03_sep2.configuration = json.dumps(conf)
```

Commit:
Populate new columns data fields

- Populate new added colums in expconf
- Fix error appearing when leaving edition mode of new columns Ref
Enabled and Ref Pattern

reszelaz commented on src/sardana/taurus/qt/qtgui/extra_sardana/measurementgroup.py:
```diff
@@ -125,6 +125,8 @@ def createChannelDict(channel, index=None, **kwargs):
         #           'timer': '', #should contain a channel name
         #           'monitor': '', #should contain a channel name
         #           'trigger': '', #should contain a channel name
+        'value_ref_enabled': True,  # bool
```
For the moment It's better to use default value false. we won't distinguish between the exp channel types neither itf these are referable or not, so it does not have too much sense to leave it empty. Just changing the default value is enough. Thanks @mrosanes!

reszelaz committed:
Change default value of value_ref_enabled to False
